sparkmagic/sparkmagic/livyclientlib/sqlquery.py

```diff
@@ -59,25 +59,28 @@ def execute(self, session):
             (success, records_text) = command.execute(session)
             if not success:
                 raise BadUserDataException(records_text)
-            result = self._records_to_dataframe(records_text)
+            result = self._records_to_dataframe(records_text, session.kind)
         except Exception as e:
             self._spark_events.emit_sql_execution_end_event(session.guid, session.kind, session.id, self.guid,
                                                             command_guid, False, e.__class__.__name__, str(e))
             raise
+
         else:
             self._spark_events.emit_sql_execution_end_event(session.guid, session.kind, session.id, self.guid,
                                                             command_guid, True, "", "")
             return result
 
     @staticmethod
-    def _records_to_dataframe(records_text):
+    def _records_to_dataframe(records_text, kind):
         if records_text == '':
             strings = []
         else:
             strings = records_text.split('\n')
         try:
             data_array = [json.JSONDecoder(object_pairs_hook=OrderedDict).decode(s) for s in strings]
-            
+
+            if kind == constants.SESSION_KIND_SPARKR and len(data_array) > 0:
+                data_array = data_array[0]
```

Comment:
suggest adding a newline for clarity, since this falls through to run the next `if` on L84